test/e2e/upgrade/upgrade.go

```diff
@@ -495,9 +509,13 @@ func IsPoolUpdated(dc dynamic.NamespaceableResourceInterface, name string) (bool
 			degraded = true
 		}
 	}
```

Comment:
do you need to handle the case where you do not find one of the `updated`, `updating`, or `degraded` condition markers?  You're defaulting the values, so if for some reason you don't find one of the conditions you're potentially going to report back a state that doesn't reflect reality.

Reply:
This is a backport, so let's take review comments that are generic to 4.8

Reply:
Cluster would be severely broken if machine pool is not either of this conditions and we'll have operator failure showing that